src/RedirectMiddleware.php

```diff
@@ -108,23 +110,26 @@ public function checkRedirect(
         if (!empty($options['allow_redirects']['track_redirects'])) {
             return $this->withTracking(
                 $promise,
-                (string) $nextRequest->getUri()
+                (string) $nextRequest->getUri(),
+                (int) $response->getStatusCode()
```

Comment:
Is there any added value in casting it to int? It will become string anyway. If ou just followed the pattern: the URI needs to be casted so that the URI object becomes string.